### ISSUE TITLE
Modify number of acceptable loss particles before termination

### ISSUE THREAD

**Issue by stevendargaville** [openmc-dev/openmc#1471]
This issue is added in response to a requirement from @Shimwell and @ukaea.

One feature we would like to add is the ability within OpenMC to specify the number of acceptable lost particles before the simulation terminates. The absolute number of lost particles and the relative number of lost particles. Currently this is hard coded within the C++ layer. We would like to have this accessible to the user (C++ and C/Python API).

I'll begin working on this shortly, but any suggestions or thoughts would be much appreciated.

**Comment by Shimwell** [comment]
Yes please this would be most handy. Absolute and relative numbers would be nice
